### PR TITLE
Generalize util method for loading class from service loader

### PR DESCRIPTION
**Pull request by bowenliang123**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

File: externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/connection/ConnectionProvider.scala [hunk 1]
```diff
@@ -69,27 +69,12 @@ abstract class AbstractConnectionProvider extends Logging {
...
-        case t: Throwable =>
-          warn(s"Loaded of the provider failed with the exception", t)
-      }
-    }
-
-    // TODO support disable provider
-    providers
-  }
+        provider
+      }.toSeq
```
Comment (bowenliang123) [inline externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/connection/ConnectionProvider.scala:77]: Could you have a look at this change? Trying to simplify the loading and looping of JdbcConnectionProvider. And skipping the warning message as there's no instance construction and no possible thrown exception.

cc @zhouyifan279

**Comment by bowenliang123** [comment]
Thanks, merged to master.
